### PR TITLE
zapier tweaks

### PR DESCRIPTION
**Pull request by snopoke**
Just some things I noticed while working on something else

### CODE REVIEW

File: corehq/apps/zapier/views.py [hunk 1]
```diff
@@ -118,7 +117,7 @@ def post(self, request, *args, **kwargs):
         user_name = request.GET.get('user')
 
         if not case_type or not owner_id or not domain or not case_name:
-            return HttpResponseForbidden('Please fill in all required fields')
+            return HttpResponseBadRequest('Please fill in all required fields')
```
Comment (snopoke) [inline corehq/apps/zapier/views.py]: changes from 403 -> 400

File: corehq/apps/zapier/views.py [hunk 2]
```diff
@@ -160,13 +159,10 @@ def post(self, request, *args, **kwargs):
         if not couch_user.is_member_of(domain):
             return HttpResponseForbidden("This user does not have access to this domain.")
 
-        try:
-            case = CaseAccessors(domain).get_case(case_id)
-        except CaseNotFound:
-            return HttpResponseForbidden("Could not find case in domain")
+        case = get_case_or_404(domain, case_id)
```
Comment (snopoke) [inline corehq/apps/zapier/views.py]: changes from 403 -> 404

File: corehq/apps/zapier/views.py [hunk 3]
```diff
@@ -160,13 +159,10 @@ def post(self, request, *args, **kwargs):
...
 
         if not case.type == case_type:
-            return HttpResponseForbidden("Case type mismatch")
+            return HttpResponseBadRequest("Case type mismatch")
```
Comment (snopoke) [inline corehq/apps/zapier/views.py]: changes from 403 -> 400

File: corehq/util/view_utils.py [hunk 4]
```diff
@@ -129,3 +133,23 @@ def reverse(viewname, params=None, absolute=False, **kwargs):
...
+            raise Http404()
+        return case
+    except CaseNotFound:
+        raise Http404()
+
+
+def get_form_or_404(domain, id):
+    try:
+        form = FormAccessors(domain).get_form(id)
+        if form.domain != domain:
```
Comment (pr33thi) [inline corehq/util/view_utils.py]: should this have a similar `form.is_deleted` check?

Comment (snopoke) [reply corehq/util/view_utils.py]: Seems reasonable: e532ad7

**Review by pr33thi — APPROVED** [review]
Travis failure looks related

**Pull request merged**